### ISSUE TITLE
Multiple Modules 

### ISSUE THREAD

**Issue by AlisonVilela** [JavadRasouli/ng2-permission#7]
I have a problem when I use `[hasPermission]="role"` in sub-modules, I imported `Ng2Permission` in my `app.module` and if I import into `another.module` for use that, is created another instance for `PermissionService` and all business rules about permissions will been lost.

@JavadRasouli do you have any suggestion?

**Comment by AlexandreKavalerski** [comment]
@AlisonVilela take a look about CoreModules here: [ngModule](https://angular.io/guide/ngmodule). I think it could solve your problem

**Comment by AlisonVilela** [comment]
Yeah @AlexandreKavalerski , I have a `core.module` but when I import `Ng2Permission` and `PermissionService` don't work in SubModules Routes

For example.: 

- core.module `Import Ng2Permission and PermissionService`
- app.module `Import core.module`
- app.routing.module `Use PermissionGuard`
- admin `Directory`
- - admin.module `Import core.module`
- - admin.routing.module `Use PermissionGuard`

In start the app, I define the roles `this.permissionService.define(this.roles)`
`PermissionService.store` in `add.routing.module` OK
`PermissionService.store` in `admin.routing.module` always empty

When I import `Ng2Permission` in `app.module` always routes works fine, including `admin.routing.module` but the directive`[hasPermission]` don't work in all pages of `admin.module`

So, when I import `Ng2Permission` in `core.module` or in `admin.module`, routing roles with `PermissionGuard` stops working because `PermissionService.store` is empty in this SubModule

I thing `PermissionService` has been created another instance when import `Ng2Permission` in another module

**Comment by JavadRasouli** [comment]
`PermissionService` is singletone and there is just one instance available in all modules. i must test your scenario at another time.

**Comment by JavadRasouli** [comment]
Only the root `AppModule` should import the `CoreModule` or `Ng2Permission`. In your code `core.module` multiple import. Once in `app.module` and again in `admin.module`. This will make several instances of the `PermissionService`.

**Comment by AlisonVilela** [comment]
@JavadRasouli Ok, but I need to import `Ng2Permission` into `CoreModule` and I need to import `CoreModule` into multiple modules

**Comment by jmparraguez-haulmer** [comment]
@AlisonVilela We have the same problem with lazy-loading module.
I fix it using this fork https://github.com/mastropinguino/ng2-permission with this instruction:
https://stackoverflow.com/questions/39672816/can-lazy-loaded-modules-share-the-same-instance-of-a-service-provided-by-their-p

**Comment by AlisonVilela** [comment]
@jmparra very interesting, @JavadRasouli if we configure `forRoot` in this way? we could then import into other modules without duplicating the service

**Comment by nhthuan** [comment]
Hi all,
I have same issue with lazy load modules also.
I see that @JavadRasouli  already update _ng2-permission.module.ts_ to add forRoot method in git repo, but yarn or npm still fetchs the old _angular2-permission-0.1.3.tgz_ package which doesn't contain implementation forRoot method (it's same even I get the link and paste to browser).
The link I tried:
_https://registry.yarnpkg.com/angular2-permission/-/angular2-permission-0.1.3.tgz_

I already try to clear yarn cache by **yarn cache clean angular2-permission** but without good result
Could you please tell me how can I use the latest implementation for angular2-permission?
Thanks in advanced,
